cvmfs/receiver/reactor.cc

```diff
@@ -463,40 +462,60 @@ CommitProcessor* Reactor::MakeCommitProcessor() {
 bool Reactor::HandleRequest(Request req, const std::string& data) {
   bool ok = true;
   std::string reply;
-  switch (req) {
-    case kQuit:
-      ok = WriteReply(fdout_, "ok");
-      break;
-    case kEcho:
-      ok = WriteReply(fdout_, std::string("PID: ") + StringifyUint(getpid()));
-      break;
-    case kGenerateToken:
-      ok &= HandleGenerateToken(data, &reply);
-      ok &= WriteReply(fdout_, reply);
-      break;
-    case kGetTokenId:
-      ok &= HandleGetTokenId(data, &reply);
-      ok &= WriteReply(fdout_, reply);
-      break;
-    case kCheckToken:
-      ok &= HandleCheckToken(data, &reply);
-      ok &= WriteReply(fdout_, reply);
-      break;
-    case kSubmitPayload:
-      ok &= HandleSubmitPayload(fdin_, data, &reply);
-      ok &= WriteReply(fdout_, reply);
-      break;
-    case kCommit:
-      ok &= HandleCommit(data, &reply);
-      ok &= WriteReply(fdout_, reply);
-      break;
-    case kError:
-      LogCvmfs(kLogReceiver, kLogSyslogErr,
-               "Reactor: unknown command received.");
-      ok = false;
-      break;
-    default:
-      break;
+  try {
```

Comment:
Do we actually still need the return code?  `HandleRequests()` is only used in `Reactor::Run`.  We can make the method `void` and put the `try {} catch {}` block around the use of the method.

Reply:
No we don't need the return code, but it is the reactor that can send back the reply to the user.

In this way we catch the error, we format it, we make a json out of it, we send it to the gateway, the gateway understand it, and send feedback to the final user.

I put this extra layer for the catch part that does the error formatting and then re-raise the exception.

Reply:
@jblomer , do we agree on this one?

We catch the code internally, we write the message to the output socket, and then we re-throw up the error for manage it at higher level.